collector/parameters.go

```diff
@@ -0,0 +1,68 @@
+package collector
+
+import (
+	"strings"
+
+	"github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	metric_mapping = map[string]prometheus.Gauge{}
+)
+
+type ParameterMetrics struct {
+}
+
+func (p *ParameterMetrics) Export(ch chan<- prometheus.Metric) {
+	for _, metric := range metric_mapping {
+		metric.Collect(ch)
+	}
+}
+
+func (p *ParameterMetrics) Describe(ch chan<- *prometheus.Desc) {
+	for _, metric := range metric_mapping {
+		metric.Describe(ch)
+	}
+}
+
+func GetParameters(session *mgo.Session, parameters string) *ParameterMetrics {
```

Comment:
Can this function be called more than once? If so `metricMapping` should be a variable of `parameterMetrics`

Reply:
This is definitely called more than once, but this is the main "collection" mechanism

Which is why metricMapping should be a global variable (as are all other metrics in this package)